blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py

```diff
@@ -122,50 +124,57 @@ def workorder_event_handler_func(self, event, block_num, txn_id, status):
         payload_string = event['payload'].decode("utf-8")
         work_order_req = json.loads(payload_string)
         work_order_id = work_order_req['workOrderId']
-        # Submit the work order to listener
-        logging.info("Submitting to work order to listener")
-        response = self.__jrpc_work_order.work_order_submit(
-            work_order_req['workOrderId'],
-            work_order_req['workerId'],
-            work_order_req['requesterId'],
-            work_order_req["workOrderRequest"],
-            id=jrpc_req_id
-        )
-        logging.info("Work order submit response {}".format(
-            response
-        ))
-        if response and 'error' in response and \
-                response['error']['code'] == WorkOrderStatus.PENDING.value:
-            # get the work order result
-            jrpc_req_id += 1
-            work_order_result = \
-                self.__jrpc_work_order.work_order_get_result(
-                    work_order_req['workOrderId'],
-                    jrpc_req_id
-                )
-            logging.info("Work order get result {}".format(
-                work_order_result
+        # Submit the work order to listener if worker id the event
+        # matches with available worker ids
+        if work_order_req['workerId'] in self.__worker_ids:
+            logging.info("Submitting to work order to listener")
+            response = self.__jrpc_work_order.work_order_submit(
+                work_order_req['workOrderId'],
+                work_order_req['workerId'],
+                work_order_req['requesterId'],
+                work_order_req["workOrderRequest"],
+                id=jrpc_req_id
+            )
+            logging.info("Work order submit response {}".format(
+                response
             ))
+            if response and 'error' in response and \
+                    response['error']['code'] == \
+                            WorkOrderStatus.PENDING.value:
+                # get the work order result
+                jrpc_req_id += 1
+                work_order_result = \
+                    self.__jrpc_work_order.work_order_get_result(
+                        work_order_req['workOrderId'],
+                        jrpc_req_id
+                    )
+                logging.info("Work order get result {}".format(
+                    work_order_result
+                ))
+            # With ZMQ based work order handler in listener returns result
+            # as a response for work order submit
+            elif response and 'result' in response:
+                work_order_result = response
```

Comment:
Please add similar change for ethereum. We could rephrase the comment "ZMQ based work order handler " to " With Synchronous work order processing" . ZMQ could be replaced with other approach for synchronous work order processing later.

Reply:
Currently ethereum is not validating work order submit response. I can add similar work order submit response validation in ethereum as well.